### PR TITLE
[Chore] Migrate to Xcode 11.4.1

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This PR includes changes to migrate from Xcode 10.2.3 to Xcode 11.3. Changes include:

- update to recommended project settings
- fix unused result warning
- fix mutability warning
- fix unused `newValue` in setter
- fix missing nullability specifiers
- fix implicit conversion warning
- fix method implemented in category when it should be implemented in the base
- update snapshot test method after API change

### Needs Release With

- [x] https://github.com/wireapp/wire-ios-shared-resources/pull/39
- [x] https://github.com/wireapp/wire-ios-system/pull/47
- [x] https://github.com/wireapp/wire-ios-testing/pull/32
- [x] https://github.com/wireapp/wire-ios-utilities/pull/88
- [x] https://github.com/wireapp/wire-ios-cryptobox/pull/35
- [x] https://github.com/wireapp/wire-ios-protos/pull/36
- [x] https://github.com/wireapp/wire-ios-transport/pull/155
- [x] https://github.com/wireapp/wire-ios-images/pull/35 (disable failed tests due to API change)
- [x] https://github.com/wireapp/wire-ios-link-preview/pull/54
- [x] https://github.com/wireapp/wire-ios-mocktransport/pull/142
- [x] https://github.com/wireapp/wire-ios-data-model/pull/871 (disable failed tests due to API change)
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/188
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/1138
- [x] https://github.com/wireapp/wire-ios-share-engine/pull/143
- [x] https://github.com/wireapp/wire-ios-canvas/pull/22
- [x] https://github.com/wireapp/Down/pull/16
- [x] https://github.com/wireapp/wire-ios-ziphy/pull/27

### To Do

- [x] Unpin azure pipeline
- [x] fix alert/action sheet snapshots
- [ ] undo tolerance to 0
- [ ] restore disabled tests  

### Notes

There are 2 build warnings remaining:
- PINCache is not safe for use in application extensions
- missing App Store icons for all non public build flavors (trivial but annoying)

### CODE REVIEW

**Comment by billypchan** [comment]
We need to bump Xcode version, ref to Xcode 10.2 ~> 10.3 PR:
https://github.com/wireapp/wire-ios-shared-resources/pull/35/files

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8807,14 +8801,69 @@
...
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
 				HEADER_SEARCH_PATHS = "$(SDKROOT)/usr/include/libxml2";
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: These flags are defined in`Warnings.xcconfig` and shared among frame projects.

**Comment by billypchan** [comment]
TODO:
- [ ] bump (or clean up) Swift version: https://github.com/wireapp/wire-ios/blob/develop/Wire-iOS/Resources/Configuration/Project-Base.xcconfig
- [ ] bump Xcode verion in https://github.com/wireapp/wire-ios/blob/develop/azure-pipelines.yml
- [ ] point wire-ios-shared-resources branch to Xcode 11 branch in https://github.com/wireapp/wire-ios/blob/develop/azure-pipelines.yml

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 1 above)
Comment (johnxnguyen) [reply Wire-iOS.xcodeproj/project.pbxproj]: Removed and cleaned up duplicates

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -8807,14 +8801,19 @@
 		8F42C556199244A700288E4D /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: This is already exist in `Warnings.xcconfig`
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -8807,14 +8801,19 @@
...
+				ENABLE_TESTABILITY = YES;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: This exists in `Project-Debug.xcconfig`
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -8807,14 +8801,19 @@
...
 				HEADER_SEARCH_PATHS = "$(SDKROOT)/usr/include/libxml2";
+				ONLY_ACTIVE_ARCH = YES;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: This exists in Project-Debug.xcconfig
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -8807,14 +8801,19 @@
...
 			};
 			name = Debug;
 		};
 		8F42C557199244A700288E4D /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]: This exists in Project-Release.xcconfig
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -8807,14 +8801,19 @@
...
 				HEADER_SEARCH_PATHS = "$(SDKROOT)/usr/include/libxml2";
+				SWIFT_COMPILATION_MODE = wholemodule;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj]:
```suggestion
```

File: Wire-iOS/Resources/Configuration/Warnings.xcconfig [hunk 7]
```diff
@@ -72,20 +69,10 @@ GCC_NO_COMMON_BLOCKS = YES
...
-CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES
 CLANG_WARN_OBJC_IMPLICIT_ATOMIC_PROPERTIES = YES
-CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES
-CLANG_WARN_OBJC_LITERAL_CONVERSION = YES
 CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR
-CLANG_WARN_STRICT_PROTOTYPES = YES
 CLANG_WARN_UNREACHABLE_CODE = YES
+
+ENABLE_STRICT_OBJC_MSGSEND = YES;
+ENABLE_TESTABILITY = YES;
```
Comment (billypchan) [inline Wire-iOS/Resources/Configuration/Warnings.xcconfig]: These lines are already exists in other xcconfig files

**Pull request merged**